### PR TITLE
feat: add testnode implementation

### PR DESCRIPTION
**Pull request by SweeXordious**
## Overview

Closes https://github.com/celestiaorg/orchestrator-relayer/issues/55

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

**Comment by evan-forbes** [comment]
wow, this is really cool!

**Review by evan-forbes — DISMISSED** [review]
nice! took a second look, and this makes sense to me so far. LGTM, utACK :+1: :+1: 

 will defer to @rahulghangas for final approval

**Comment by SweeXordious** [comment]
CI will be fixed by:  https://github.com/celestiaorg/orchestrator-relayer/pull/101

**Review by evan-forbes — APPROVED** [review]

**Comment by SweeXordious** [comment]
@rahulghangas Can you please check this so that I can open up new PRs?

File: testing/evm_chain.go
Comment (SweeXordious) [reply testing/evm_chain.go:24]: I have docs for that in local. After this is merged, new PRs containing that will come. Thanks for flagging this

File: testing/evm_chain.go
Comment (SweeXordious) [reply testing/evm_chain.go:69]: https://github.com/celestiaorg/orchestrator-relayer/pull/95#discussion_r1101178234

File: testing/testnode.go
Comment (SweeXordious) [reply testing/testnode.go:47]: https://github.com/celestiaorg/orchestrator-relayer/pull/95#discussion_r1101178234

**Pull request merged**
